src/typescript-worker/typescript-worker.ts

```diff
@@ -0,0 +1,352 @@
+/**
+ * @license
+ * Copyright (c) 2019 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {TypeScriptWorkerAPI, SampleFile} from '../shared/worker-api.js';
+import {expose} from 'comlink';
+import * as ts from 'typescript';
+
+const compilerOptions = {
+  target: ts.ScriptTarget.ES2017,
+  module: ts.ModuleKind.ESNext,
+  experimentalDecorators: true,
+  skipDefaultLibCheck: true,
+  skipLibCheck: true,
+  moduleResolution: ts.ModuleResolutionKind.NodeJs,
+};
+
+/**
+ * Rewrites bare module specifiers to unpkg.com URLs. For now, uses unpkg.com
```

Comment:
unpkg module resolution fails when you point to something like

```
import '@material/mwc-checkbox';
import '@material/mwc-list/mwc-list-item.js`;
```

in this case the paths are weird in such that pointing to a specific file causes a duplication of the `@material/mwc-ripple` file.